detection_rules/rule.py

```diff
@@ -515,6 +515,55 @@ def name(self) -> str:
     def type(self) -> str:
         return self.data.type
 
+    def _post_dict_transform(self, obj: dict) -> dict:
+        """Derive required fields from query AST."""
+
+        super()._post_dict_transform(obj)
+
+        self.add_related_integrations(obj)
+        self.add_required_fields(obj)
+        self.add_setup(obj)
+
+        return obj
+
+    def add_related_integrations(self, obj: dict) -> None:
+        """Add restricted field add_related_integrations to the obj"""
+        # field_name = "related_integrations"
+        ...
+
+    def add_required_fields(self, obj: dict) -> None:
+        """Add restricted field add_required_fields to the obj"""
+
+        field_name = "required_fields"
+        field_value = obj.get(field_name, [])
+        if self.check_restricted_field_version(field_name):
+
+            # validate the built field
+            self.validate_restricted_field_type(field_name, field_value)
+            obj.setdefault(field_name, field_value)
```

Comment:
a simpler and more accurate approach would be to just load the actual data

```suggestion
            obj.setdefault(field_name, field_value)
```

like this:
```python
            # validate new field against the schema
            rule_type = obj['type']
            subclass =  self.get_data_subclass(rule_type)
            subclass.from_dict(obj)
```

But I am going to add this section to the transform method so that we only do this once, _after_ all of the build fields to avoid doing it repetitively